internal/network/manifests_generator.go

```diff
@@ -509,3 +511,71 @@ func NewConfig() (*Config, error) {
 	}
 	return &networkCfg, nil
 }
+
+const nodeIpHint = `
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: {{.ROLE}}
+  name: 10-{{.ROLE}}s-node-ip-hint
+spec:
+  config:
+    ignition:
+      config: {}
+      security:
+        tls: {}
+      timeouts: {}
+      version: 2.2.0
+    networkd: {}
+    passwd: {}
+    storage:
+      files:
+      - contents:
+          source: data:text/plain;charset=utf-8;base64,{{.NODE_IP_CONTENT}}
+          verification: {}
+        filesystem: root
+        mode: 420
+        path: /etc/default/nodeip-configuration
+`
+
+// Add node ip hint is supported from 4.11 and will allow us to tell to node-ip script which ip kubelet should run with
+// https://github.com/openshift/machine-config-operator/commit/a0c9a3caa54018eb89eb5bdd6ec1b8fbf97f6fb7
+func (m *ManifestsGenerator) AddNodeIpHint(ctx context.Context, log logrus.FieldLogger, cluster *common.Cluster) error {
+	if hintSupported, err := common.VersionGreaterOrEqual(cluster.OpenshiftVersion, "4.11.0-0.alpha"); err != nil {
+		return err
+	} else if !hintSupported {
+		return nil
+	}
+
+	filename := "node-ip-hint.yaml"
+
+	content, err := createNodeIpHintContent(log, cluster)
+	if err != nil {
+		log.WithError(err).Errorf("Failed to create node ip hint manifest")
+		return err
+	}
+
+	return m.createManifests(ctx, cluster, filename, content)
+}
+
+func createNodeIpHintContent(log logrus.FieldLogger, cluster *common.Cluster) ([]byte, error) {
+	if !IsMachineCidrAvailable(cluster) {
+		return nil, fmt.Errorf("machine networks should be configured")
+	}
+	machineCidr := cluster.MachineNetworks[0]
```

Comment:
I don't think we need to address this now but I do wonder how it would work with dual stack

Reply:
should be the same , that's why i am using [0] as it will be always primary one